### PR TITLE
Kevin/live questions

### PR DESCRIPTION
**Pull request by kevinsun-dev**
## Overview

Added live question functionality, full socket support for polling, and revamped a few models.

  ## Changes Made
Redid Sockets.kt models and networking, added functions to interface with existing live poll UI.

  ## Next Steps
Lots of issues still in Live Polling, doesn't automatically show results when poll finishes, crashes a few times, drops out of groups occasionally, etc. Requires a fair bit of cleanup and refactoring.

### CODE REVIEW

**Comment by kevinsun-dev** [comment]
This PR will be massive, unclean, and rushed. I'll demo screenshots and types at the next meeting with the help of iOS, but I'm dropping this PR in time for Demo day.

File: app/src/main/java/com/cornellappdev/android/pollo/Polls/PollsActivity.kt [hunk 1]
```diff
@@ -24,22 +38,112 @@ class PollsActivity : AppCompatActivity() {
...
+        polls = intent.getParcelableArrayListExtra<Poll>("POLLS")
         name = intent.getStringExtra("GROUP_NAME")
         code = intent.getStringExtra("GROUP_CODE")
+        date = intent.getStringExtra("POLLS_DATE")
+        userCount = intent.getIntExtra("USER_COUNT", 0)
+        Socket.add(this)
 
         groupNameTextView.text = name
         codeTextView.text = "Code: $code"
+//        userCountTextView.text = "$userCount"
```
Comment (jmf373) [inline app/src/main/java/com/cornellappdev/android/pollo/Polls/PollsActivity.kt]: Can we delete this/other `userCount` code? I think design removed the user count view but I could be wrong

File: app/src/main/java/com/cornellappdev/android/pollo/Polls/PollsActivity.kt [hunk 2]
```diff
@@ -24,22 +38,112 @@ class PollsActivity : AppCompatActivity() {
...
         pollsRecyclerView.layoutManager = linearLayoutManager
-        adapter = PollsRecyclerAdapter(polls)
+        adapter = PollsRecyclerAdapter(polls, googleId)
         pollsRecyclerView.adapter = adapter
 
+        if (polls[polls.size - 1].state == PollState.live) {
+            linearLayoutManager.scrollToPosition(polls.size - 1)
+        }
+
+
```
Comment (jmf373) [inline app/src/main/java/com/cornellappdev/android/pollo/Polls/PollsActivity.kt]: Nit: remove newlines

File: app/src/main/java/com/cornellappdev/android/pollo/Polls/PollsRecyclerAdapter.kt [hunk 3]
```diff
@@ -41,27 +58,43 @@ class PollsRecyclerAdapter(private val polls: ArrayList<PollsResponse>) : Recycl
             val endAdjustment = (widthSubtraction / 2) - fudgeFactor
             marginStart = if (position == 0) endAdjustment else 16
             marginEnd = if (position == (itemCount - 1)) endAdjustment else 16
+
+            val headerHeight = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 100f, displayMetrics).toInt()
+            val cellHeight = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 55f, displayMetrics).toInt()
+            //86 dp is the heght of header, 53dp is height of cell
+            val tmpHeight = headerHeight + cellHeight*poll.answerChoices.count() //53 is cell height including top margin
+
+            height = if (tmpHeight <= 1250 && poll.type == PollType.multipleChoice) tmpHeight else 1250
```
Comment (jmf373) [inline app/src/main/java/com/cornellappdev/android/pollo/Polls/PollsRecyclerAdapter.kt]: Can we have a comment here explaining this?

**Pull request merged**